embulk-input-mysql/src/main/java/org/embulk/input/mysql/getter/MySQLDateTimeTimestampIncrementalHandler.java

```diff
@@ -38,10 +38,10 @@ public String getTimestampPattern()
     }
 
     @Override
-    public Timestamp utcTimestampToSessionTime(org.embulk.spi.time.Timestamp from)
+    public Timestamp utcTimestampToSessionTime(final Instant from)
     {
         // reconvert from UTC to session time_zone
-        final long reconverted = ZonedDateTime.ofInstant(Instant.ofEpochSecond(from.getEpochSecond()), ZoneOffset.UTC)
+        final long reconverted = ZonedDateTime.ofInstant(from, ZoneOffset.UTC)
```

Comment:
i think this code block qual to `return Timestamp.from(from.atZone(sessionTimeZone).toInstant());`

Reply:
Hmm, I don't think so. `from.atZone(sessionTimeZone).toInstant()` results in `Instant` same with `from`. The `withZoneSameLocal` method does the tricky computation.